### PR TITLE
Dsl2

### PR DESCRIPTION
**Pull request by marangiop**
Making pull request for migration to DSL2 within context of PrecisionTox

### CODE REVIEW

File: grape-pipeline.nf [hunk 1]
```diff
@@ -46,20 +48,51 @@ params.wigRefPrefix = ''
...
-    txIndexGenomes; txIndexAnnotations
-    mergeBamInput
-    markdupInput
-    inferExpInput; inferExpAnnotations
-    bamStatsInput; bamStatsAnnotations
-    bigwigInput; bigwigFastaIndex
-    contigInput; contigFastaIndex
-    quantificationInput; quantificationTxIndex
-}
+Channel.empty().set { fetchInput }
```
Comment (emi80) [inline grape-pipeline.nf:51]: we should check if all these channel initialisations are still needed with DSL2

File: grape-pipeline.nf [hunk 2]
```diff
@@ -255,17 +288,17 @@ if ( 'mapping' in pipelineSteps ) {
...
     } else {
-        genomesForIndex.set { indexGenomes }
-        annotationsForIndex.set { indexAnnotations }
+        Genomes.set { indexGenomes }
+        Annotations.set { indexAnnotations }
     }
-    annotationsForMapping.set { mappingAnnotations }
-    annotationsForInferExp.set { inferExpAnnotations }
-    annotationsForBamStats.set { bamStatsAnnotations }
+    Annotations.set { mappingAnnotations }
```
Comment (emi80) [inline grape-pipeline.nf:294]: I guess we could get rid of all duplicated channels as channels can be used multiple times in DSL2

File: grape-pipeline.nf [hunk 3]
```diff
@@ -578,26 +539,27 @@ process markdup {
...
 
-if ( params.markDuplicates || params.removeDuplicates ) {
-    bamFilesToGenome = markdupOutput
-}
 
-bamFilesToGenome.into {
-    inferExpInput
-    bamStatsInput
-}
+
```
Comment (emi80) [inline grape-pipeline.nf:543]: I would clean up commented old code like this

File: grape-pipeline.nf (hunk 1 above)
Comment (marangiop) [reply grape-pipeline.nf:51]: So I suppose the channel initialisations are still needed

File: grape-pipeline.nf [hunk 4]
```diff
@@ -292,70 +327,45 @@ process fetch {
...
 
-if ( 'bigwig' in pipelineSteps ) {
-    fastaIndexForBigwig.set { bigwigFastaIndex }
-}
-
-if ( 'contig' in pipelineSteps ) {
-    fastaIndexForContig.set { contigFastaIndex }
-}
 
 process index {
```
[the remark this answers is not in the document]
Comment (marangiop) [reply grape-pipeline.nf]: Done

File: grape-pipeline.nf (hunk 3 above)
Comment (marangiop) [reply grape-pipeline.nf]: Done

File: grape-pipeline.nf (hunk 2 above)
Comment (marangiop) [reply grape-pipeline.nf]: Done

**Pull request merged**